lizmap/plugin.py

```diff
@@ -2161,9 +2161,17 @@ def set_tree_item_data(self, item_type, item_key, json_layers):
         # DEFAULT VALUES : layers have got more precise data
         keep_metadata = False
         if item_type == 'layer':
+            layer = self.get_qgis_layer_by_id(item_key)
+            
+            # layer corrupted ?
+            if layer is None:
+                error_msg = tr(
+                    "The layer '{}' seems invalid. Check the layer configuration."
+                ).format(item_key)
+                self.display_error(error_msg)
+                raise Exception(error_msg)
```

Comment:
```suggestion
                return
```

Raising an exception (moreover a generic one such as "Exception") without catching it is not a good user experience, as it's raising dialog with some Python.

I think you have encountered what we call "ghost layers".
https://github.com/3liz/lizmap-plugin/blob/b82f43e500f89be77dd551c345063c672c111624/lizmap/toolbelt/layer.py#L69 Or it seems the reverse for you ?

Reply:
@Gustry I have the reverse of this, a layer that show in the tree but that is not functional due to the missing layer definition
This is probably explained by save on the project using a QGIS version not able to read this layers